Kudu.Core/Permissions/PermissionHandler.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.IO;
+using Kudu.Contracts.Permissions;
+using Kudu.Contracts.Settings;
+using Kudu.Core.Deployment;
+using Kudu.Core.Deployment.Generator;
+using Kudu.Core.Infrastructure;
+
+namespace Kudu.Core.Permissions
+{
+    /// <summary>
+    /// This handler is meant for Linux enviroment only
```

Comment:
If the intention is simply Chmod of the id_rsa and its associates (to workaround certain OS), having IPermissionHandler might be a bit overkilled (if I read the code correctly).    We could something along this line.

1) to back fill old sites, you could do similarly [here](https://github.com/projectkudu/kudu/blob/6886065d11e1f3b37f52dcc65a69c7eb4de3d95d/Kudu.Services.Web/App_Start/NinjectServices.cs#L552). 

2) For new file, also do it where we initially create [the id_rsa file](https://github.com/projectkudu/kudu/blob/6886065d11e1f3b37f52dcc65a69c7eb4de3d95d/Kudu.Core/SSHKey/SSHKeyManager.cs).

Is this class useful for other scenarios?